ui/jobs/components/sam.js

```diff
@@ -1,6 +1,97 @@
 import EffectId from '../../../resources/effect_id.js';
+import { kAbility } from '../constants.js';
 
 export function setup(bars) {
   bars.onYouGainEffect(EffectId.Shifu, () => bars.speedBuffs.shifu = 1);
   bars.onYouLoseEffect(EffectId.Shifu, () => bars.speedBuffs.shifu = 0);
+
+  const comboTimer = bars.addTimerBar({
+    id: 'sam-timers-combo',
+    fgColor: 'combo-color',
+  });
+  bars.onCombo((skill) => {
+    comboTimer.duration = 0;
+    if (bars.combo.isFinalSkill)
+      return;
+    if (skill)
+      comboTimer.duration = 15;
+  });
+
+  const kenkiGauge = bars.addResourceBox({
+    classList: ['sam-color-kenki'],
+  });
+  const meditationGauge = bars.addResourceBox({
+    classList: ['sam-color-meditation'],
+  });
+  bars.onJobDetailUpdate((jobDetail) => {
+    kenkiGauge.innerText = jobDetail.kenki;
+    meditationGauge.innerText = jobDetail.meditationStacks;
+    if (jobDetail.kenki >= 70)
+      kenkiGauge.parentNode.classList.add('high');
+    else
+      kenkiGauge.parentNode.classList.remove('high');
+    if (jobDetail.meditationStacks >= 2)
+      meditationGauge.parentNode.classList.add('high');
+    else
+      meditationGauge.parentNode.classList.remove('high');
+  });
+
+  const shifu = bars.addProcBox({
+    id: 'sam-procs-shifu',
+    fgColor: 'sam-color-shifu',
+  });
+  bars.onYouGainEffect(EffectId.Shifu, (name, matches) => {
```

Comment:
There is a `onYouGainEffect` with shifu above too, it would be overrided if you call this function twice. So you either write down the shifu asignment here, otherwise merge these function into the function above.

In fact, the parameters of the function are **ID** and Matches, although it is not being used here.